experimental/dds/tree2/src/feature-libraries/chunked-forest/chunkedForest.ts

```diff
@@ -142,43 +121,82 @@ class ChunkedForest extends SimpleDependee implements IEditableForest {
+			destroy(detachedField: FieldKey, count: number): void {
+				this.forest.invalidateDependents();
+				const parent = this.getParent();
+				parent.mutableChunk.fields.delete(detachedField);
 			},
-			onInsert(index: number, content: Delta.ProtoNodes): void {
+			create(content: Delta.ProtoNodes, destination: FieldKey): void {
 				this.forest.invalidateDependents();
 				const chunks: TreeChunk[] = content.map((c) => chunkTree(c, this.forest.chunker));
-				const field = this.forest.newDetachedField();
-				this.forest.roots.fields.set(detachedFieldAsKey(field), chunks);
-				this.moveIn(index, field, false);
+				this.forest.roots.fields.set(destination, chunks);
+			},
+			attach(source: FieldKey, count: number, destination: PlaceIndex): void {
+				this.forest.invalidateDependents();
+				this.attachEdit(source, count, destination);
 			},
-			onMoveOut(index: number, count: number, id?: Delta.MoveId): void {
+			detach(source: Range, destination: FieldKey): void {
 				this.forest.invalidateDependents();
+				this.detachEdit(source, destination);
+			},
+			/**
+			 * Attaches the range into the current field by transferring it from the given source path.
+			 * Does not invalidate dependents.
+			 * @param source - The the range to be attached.
+			 * @param destination - The index in the current field at which to attach the content.
+			 */
+			attachEdit(source: FieldKey, count: number, destination: PlaceIndex): void {
+				const sourceField = this.forest.roots.fields.get(source) ?? [];
+				this.forest.roots.fields.delete(source);
+				if (sourceField.length === 0) {
+					return; // Prevent creating 0 sized fields when inserting empty into empty.
+				}
+
+				const parent = this.getParent();
+				const destinationField = getOrAddEmptyToMap(parent.mutableChunk.fields, parent.key);
+				// TODO: this will fail for very large moves due to argument limits.
+				destinationField.splice(destination, 0, ...sourceField);
+			},
+			/**
+			 * Detaches the range from the current field and transfers it to the given destination if any.
+			 * Does not invalidate dependents.
+			 * @param source - The bounds of the range to be detached from the current field.
+			 * @param destination - If specified, the destination to transfer the detached range to.
+			 * If not specified, the detached range is destroyed.
+			 */
+			detachEdit(source: Range, destination: FieldKey | undefined): void {
 				const parent = this.getParent();
 				const sourceField = parent.mutableChunk.fields.get(parent.key) ?? [];
-				const newField = sourceField.splice(index, count);
 
-				if (id !== undefined) {
-					const detached = this.forest.newDetachedField();
-					const key = detachedFieldAsKey(detached);
+				assertValidRange(source, sourceField);
+				const newField = sourceField.splice(source.start, source.end - source.start);
+
+				if (sourceField.length === 0) {
+					parent.mutableChunk.fields.delete(parent.key);
+				}
+				if (destination !== undefined) {
```

Comment:
Since we now require the destination to be a new empty detached field, there should be an assert here that this.forest.roots.fields.has(destintion) is false.